test/test_json_commands.py

```diff
@@ -0,0 +1,1815 @@
+    assert r.json().type("doc1", "$..a") == jtypes
+
+    # Test single
+    assert r.json().type("doc1", "$.nested2.a") == [jtypes[1]]
+
+    # Test missing key
+    assert r.json().type("non_existing_doc", "..a") is None
+
+
+@pytest.mark.xfail
+def test_clear_dollar(r: redis.Redis) -> None:
+    r.json().set(
+        "doc1",
+        "$",
+        {
+            "nested1": {"a": {"foo": 10, "bar": 20}},
+            "a": ["foo"],
+            "nested2": {"a": "claro"},
+            "nested3": {"a": {"baz": 50}},
+        },
+    )
+    # Test multi
+    assert r.json().clear("doc1", "$..a") == 3
+
+    assert r.json().get("doc1", "$") == [
+        {"nested1": {"a": {}}, "a": [], "nested2": {"a": "claro"}, "nested3": {"a": {}}}
+    ]
+
+    # Test single
+    r.json().set(
+        "doc1",
+        "$",
+        {
+            "nested1": {"a": {"foo": 10, "bar": 20}},
+            "a": ["foo"],
+            "nested2": {"a": "claro"},
+            "nested3": {"a": {"baz": 50}},
+        },
+    )
+    assert r.json().clear("doc1", "$.nested1.a") == 1
+    assert r.json().get("doc1", "$") == [
+        {
+            "nested1": {"a": {}},
+            "a": ["foo"],
+            "nested2": {"a": "claro"},
+            "nested3": {"a": {"baz": 50}},
+        }
+    ]
+
+    # Test missing path (defaults to root)
+    assert r.json().clear("doc1") == 1
+    assert r.json().get("doc1", "$") == [{}]
+
+    # Test missing key
+    with pytest.raises(exceptions.ResponseError):
+        r.json().clear("non_existing_doc", "$..a")
+
+
+@pytest.mark.xfail
+def test_toggle_dollar(r: redis.Redis) -> None:
+    r.json().set(
+        "doc1",
+        "$",
+        {
+            "a": ["foo"],
+            "nested1": {"a": False},
+            "nested2": {"a": 31},
+            "nested3": {"a": True},
+        },
+    )
+    # Test multi
+    assert r.json().toggle("doc1", "$..a") == [None, 1, None, 0]
+    assert r.json().get("doc1", "$") == [
+        {
+            "a": ["foo"],
+            "nested1": {"a": True},
+            "nested2": {"a": 31},
+            "nested3": {"a": False},
+        }
+    ]
+
+    # Test missing key
+    with pytest.raises(exceptions.ResponseError):
+        r.json().toggle("non_existing_doc", "$..a")
+
+
+# # @pytest.mark.xfail
+# def test_debug_dollar(r: redis.Redis) -> None:
+#
+#    jdata, jtypes = load_types_data("a")
+#
+#    r.json().set("doc1", "$", jdata)
+#
+#    # Test multi
+#    assert r.json().debug("MEMORY", "doc1", "$..a") == [72, 24, 24, 16, 16, 1, 0]
+#
+#    # Test single
+#    assert r.json().debug("MEMORY", "doc1", "$.nested2.a") == [24]
+#
+#    # Test legacy
+#    assert r.json().debug("MEMORY", "doc1", "..a") == 72
+#
+#    # Test missing path (defaults to root)
+#    assert r.json().debug("MEMORY", "doc1") == 72
+#
+#    # Test missing key
+#    assert r.json().debug("MEMORY", "non_existing_doc", "$..a") == []
+
+
+@pytest.mark.xfail
+def test_resp_dollar(r: redis.Redis) -> None:
+    data = {
```

Comment:
can we simplify this test?
`json.dumps`/`json.loads`? Using the same data structure in multiple tests?

Reply:
We probably can, yes. I'll break out the static stat to some kind of fixture as I work my way through implementing the functions whose tests are currently marked `xfail`.